### PR TITLE
Bind schemas to factory files, add support for groups in VPC-SC schema

### PR DESCRIPTION
**Pull request by wiktorn**
Groups and principals are [available in preview](https://cloud.google.com/vpc-service-controls/docs/release-notes#July_31_2024)

Inspired by #2489, though this means that JSON schema is more strict than `variables.tf`

---
**Checklist**

I applicable, I acknowledge that I have:
- [ ] Read the [contributing guide](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/master/CONTRIBUTING.md)
- [ ] Ran `terraform fmt` on all modified files
- [ ] Regenerated the relevant README.md files using [`tools/tfdoc.py`](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/master/CONTRIBUTING.md#fabric-tools)
- [ ] Made sure all relevant tests pass

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -6,5 +6,25 @@
...
-  ]
-}
\ No newline at end of file
+  ],
+  "yaml.schemas": {
+    "modules/project-factory/schemas/budget.schema.json": [
+      "budgets/**/*yaml"
+    ],
+    "modules/project-factory/schemas/folder.schema.json": [
+      "folders/**/*yaml"
```
Comment (ludoo) [inline .vscode/settings.json:15]: this should be either `folders` or `hierarchy`

**Comment by ludoo** [comment]
Ca we retrofit the validation to variables.tf in both vpc-sc module and FAST stage?

**Review by ludoo — APPROVED** [review]

File: .vscode/settings.json [hunk 2]
```diff
@@ -6,5 +6,25 @@
       ],
       "url": "http://json-schema.org/draft-07/schema#"
     }
-  ]
-}
\ No newline at end of file
+  ],
+  "yaml.schemas": {
```
Comment (ludoo) [inline .vscode/settings.json:10]: I would use a `data/` prefix for values, so as to pinpoint factory data.

**Comment by ludoo** [comment]
`yaml.schemas` does not seem to work
image

**Comment by wiktorn** [comment]
> `yaml.schemas` does not seem to work
> 
> image

Are you running multi-root workspace? Then [this may be applicable](https://github.com/redhat-developer/yaml-language-server?tab=readme-ov-file#multi-root-schema-association), and you need specify `yaml.schemas` in  root of the multi root workspace project.

File: .vscode/settings.json (hunk 2 above)
Comment (wiktorn) [reply .vscode/settings.json:10]: Added `data/**/` prefix, so we can have `data/<stage-name>/<object_class>` structure when pointing to directory in `factories_config`.

Comment (ludoo) [reply .vscode/settings.json:10]: `stage-name` is not needed, paths will always be `data/object-class` for both modules and FAST :)

Comment (ludoo) [reply .vscode/settings.json:10]: well I lied, I point FAST factories for my org to my own data folder which is structured exactly like you wrote 😊, so we should support both `data/object-class` and `data/stage/object-class`

Comment (wiktorn) [reply .vscode/settings.json:10]: both works with `data/**/`

Comment (ludoo) [reply .vscode/settings.json:10]: yep, absolutely that was the intention of my last message

**Pull request merged**